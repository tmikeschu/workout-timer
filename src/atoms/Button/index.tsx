import * as React from "react";
import { styled } from "theme";

const Container = styled.button`
  background-color: ${({ theme }): string => theme.primary};
  color: ${({ theme }): string => theme.primaryAlt};
  border: none;
  padding: 0.5rem 1rem;
  font-family: inherit;
  font-size: 1.5rem;
  margin-bottom: 1rem;
  width: 100%;
  max-width: 414px;

  &:last-of-type {
    margin-bottom: 0;
  }

  &:hover {
    cursor: pointer;
  }

  &:hover,
  &:focus {
    filter: brightness(0.75);
  }

  &:disabled {
    opacity: 0.5;
    filter: brightness(1);
    cursor: default;
  }
`;

const Button: React.FC<React.HTMLProps<HTMLButtonElement>> = ({
  type: _,
  ...props
}) => {
  return <Container {...props} />;
};

export default Button;
