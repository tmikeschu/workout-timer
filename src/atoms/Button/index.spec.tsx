import React from "react";
import { render } from "@testing-library/react";
import Button from "./";

describe("Button", () => {
  it("has correct snapshot", () => {
    const tree = render(<Button />).container;
    expect(tree).toMatchSnapshot();
  });
});
