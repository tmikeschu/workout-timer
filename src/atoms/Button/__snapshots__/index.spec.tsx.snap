// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button has correct snapshot 1`] = `
<div>
  <button
    class="css-twz5y6"
  />
</div>
`;
